File: cs/Iris/Data/IrisSound.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.IO;

namespace Iris.Data
{
    public class IrisSound
    {
        private static readonly string baseFolder = Directory.GetCurrentDirectory() + "\\sounds\\";

        public Uri soundPath;
        public string name;

        public IrisSound()
        {

        }

        public IrisSound(string filepath)
        {
            soundPath = new Uri(baseFolder + filepath, UriKind.Absolute);
        }

        public IrisSound(string soundName, string filepath)
            : this(filepath)
        {
            name = soundName;
        }

        public void setSource(string filepath)
        {
            soundPath = new Uri(baseFolder + filepath, UriKind.Absolute);
        }
    }
}

File: java/Iris/src/com/rokuan/iris/data/IrisSound.java
package com.rokuan.iris.data;

public class IrisSound extends IrisFile
{
	private static final String baseFolder = "sounds/";
	
	public String name;

	public IrisSound()
	{

	}

	public IrisSound(String filepath)
	{
		this.setFilepath(filepath);
	}

	public IrisSound(String soundName, String filepath)
	{
		this(filepath);
		this.name = soundName;
	}

	@Override
	public void setFilepath(String path){
		super.setFilepath(baseFolder + path);
	}
}
